### ISSUE TITLE
Windows release build shows a blank screen

### ISSUE THREAD

**Issue by TeixeiraFillipe** [Sharcoux/rn-css#27]
So, we're having an issue when using rn-css and trying to build an app for release on Windows. In debug it builds fine, no problems, but when building for release, all we get is a blank screen and no errors from the output. We also with tried different versions of rn-css, but no success. 
Any ideas of what could be causing this? Or anyone facing this same issue managed to come up with a solution?

This is my code:

```
import React from 'react';
import {View} from 'react-native';
import styled from 'rn-css';

const Body03 = styled.Text`
    font-family: AvenirNextLTPro-Regular;
    color: #000;
    text-align: center;
    font-size: 12px;
    line-height: 16px;
  `;

const App = () => {

  return (
    <View>
      <Body03>Hello World rn-css</Body03>
    </View>
  );
};
```

Screen prints from respectively debug and release build:
![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/102185908/165379228-e2b06f62-98e3-4437-aaff-f8198d29b740.png)
![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/102185908/165379301-4bbe0bdf-b7e0-40f6-845d-67427e5bb079.png)

**Comment by Sharcoux** [comment]
This is really weird. We also build apps for windows and windows store. What do you build the app with?

**Comment by TeixeiraFillipe** [comment]
We're using Visual Studio 2019, version 16.11.12

**Comment by Sharcoux** [comment]
Ok but I mean how do you transform your javascript code into a Windows native application?

**Comment by Sharcoux** [comment]
Also, have you tried to build with another lib, like style-components/native, to be sure that it's not the way your build process handles dependencies that have to be blamed?

**Comment by TeixeiraFillipe** [comment]
We're using react native for Windows (https://microsoft.github.io/react-native-windows/docs/getting-started).
And yes, with style-components/native it works, it builds fine in debug and release. It just shows that black screen when using rn-css.

**Comment by Sharcoux** [comment]
Interesting. We have never built for windows using react-native. We are using Electron for that. I'll investigate. Can you check if version 1.5.5 is working fine?

**Comment by Sharcoux** [comment]
I'll also need your full package.json if possible.

**Comment by TeixeiraFillipe** [comment]
I'll run a test with version 1.5.5 right now.
Here's my package.json:
```
{
  "name": "newproject",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "start": "react-native start",
    "test": "jest",
    "lint": "eslint .",
    "windows": "react-native run-windows"
  },
  "dependencies": {
    "react": "17.0.2",
    "react-native": "0.68.0",
    "react-native-windows": "0.68.2",
    "rn-css": "^1.6.13"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9",
    "@babel/runtime": "^7.12.5",
    "@react-native-community/eslint-config": "^2.0.0",
    "babel-jest": "^26.6.3",
    "eslint": "7.14.0",
    "jest": "^26.6.3",
    "metro-config": "^0.66.2",
    "metro-react-native-babel-preset": "^0.66.2",
    "react-test-renderer": "17.0.2"
  },
  "jest": {
    "preset": "react-native"
  }
}

```

**Comment by TeixeiraFillipe** [comment]
Just tried with version 1.5.5,  but got the same result, works on debug however in release I get the black screen.

**Comment by Sharcoux** [comment]
Ok. I'll need your help to debug this as I don't have windows nor ever generated a react-native-windows app.

**Comment by Sharcoux** [comment]
Basically, I will provide you some versions of the file `styleComponent.js` to try out. You will have to replace the content of `node_modules/rn-css/dist/styleComponent.js` by the content I give you and then try to generate the app.

Here is the first one to try:

```js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.styledVirtualizedList = exports.styledSectionList = exports.styledFlatList = exports.SharedValue = exports.FontSizeContext = exports.RemContext = exports.defaultUnits = void 0;
/* eslint-disable react/prop-types */
/* eslint-disable react/display-name */
const react_1 = __importDefault(require("react"));
const react_native_1 = require("react-native");
exports.defaultUnits = { em: 16, vw: 1, vh: 1, vmin: 1, vmax: 1, rem: 16, px: 1, pt: 72 / 96, in: 96, pc: 9, cm: 96 / 2.54, mm: 96 / 25.4 };
exports.RemContext = react_1.default.createContext(exports.defaultUnits.rem);
exports.FontSizeContext = react_1.default.createContext(exports.defaultUnits.em);
// We use this to share value within the component (Theme, Translation, whatever)
exports.SharedValue = react_1.default.createContext(undefined);
const styled = (Component) => {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    const styledComponent = (_chunks, ..._functs) => {
        const ForwardRefComponent = react_1.default.forwardRef((props, ref) => {
            return react_1.default.createElement(Component, { ref: ref, ...props });
        });
        return ForwardRefComponent;
    };
    // provide styled(Comp).attrs({} | () => {}) feature
    styledComponent.attrs = (opts) => (chunks, ...functs) => {
        const ComponentWithAttrs = styledComponent(chunks, ...functs);
        const ForwardRefComponent = react_1.default.forwardRef((props, ref) => {
            const attrs = (opts instanceof Function) ? opts(props) : opts;
            return react_1.default.createElement(ComponentWithAttrs, { ref: ref, ...props, ...attrs });
        });
        // TODO : Find a way to remove from the Props the properties affected by opts
        return ForwardRefComponent;
    };
    return styledComponent;
};
exports.default = styled;
const styledFlatList = (chunks, ...functs) => (props) => invoke(styled(react_native_1.FlatList)(chunks, ...functs), props);
exports.styledFlatList = styledFlatList;
exports.styledFlatList.attrs = (opts) => (chunks, ...functs) => (componentProps) => invoke(styled(react_native_1.FlatList).attrs(opts)(chunks, ...functs), componentProps);
const styledSectionList = (chunks, ...functs) => (props) => invoke(styled(react_native_1.SectionList)(chunks, ...functs), props);
exports.styledSectionList = styledSectionList;
exports.styledSectionList.attrs = (opts) => (chunks, ...functs) => (componentProps) => invoke(styled(react_native_1.SectionList).attrs(opts)(chunks, ...functs), componentProps);
const styledVirtualizedList = (chunks, ...functs) => (props) => invoke(styled(react_native_1.VirtualizedList)(chunks, ...functs), props);
exports.styledVirtualizedList = styledVirtualizedList;
exports.styledVirtualizedList.attrs = (opts) => (chunks, ...functs) => (componentProps) => invoke(styled(react_native_1.VirtualizedList).attrs(opts)(chunks, ...functs), componentProps);
function invoke(Component, props) {
    return react_1.default.createElement(Component, { ...props });
}
```

Of course, if you succeed to have the log of why the app crashes, we would save a lot of time...

**Comment by Sharcoux** [comment]
If the previous version worked, here is the second version to try out:

```js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.styledVirtualizedList = exports.styledSectionList = exports.styledFlatList = exports.SharedValue = exports.FontSizeContext = exports.RemContext = exports.defaultUnits = void 0;
/* eslint-disable react/prop-types */
/* eslint-disable react/display-name */
const react_1 = __importDefault(require("react"));
const react_native_1 = require("react-native");
const convertStyle_1 = __importDefault(require("./convertStyle"));
const cssToRN_1 = __importDefault(require("./cssToRN"));
const generateHash_1 = __importDefault(require("./generateHash"));
exports.defaultUnits = { em: 16, vw: 1, vh: 1, vmin: 1, vmax: 1, rem: 16, px: 1, pt: 72 / 96, in: 96, pc: 9, cm: 96 / 2.54, mm: 96 / 25.4 };
exports.RemContext = react_1.default.createContext(exports.defaultUnits.rem);
exports.FontSizeContext = react_1.default.createContext(exports.defaultUnits.em);
// We use this to share value within the component (Theme, Translation, whatever)
exports.SharedValue = react_1.default.createContext(undefined);
function buildCSSString(chunks, functs, props, shared) {
    let computedString = chunks.map((chunk, i) => ([chunk, (functs[i] instanceof Function) ? functs[i]({ shared, theme: shared, ...props }) : functs[i]])).flat().join('');
    if (props.rnCSS)
        computedString += props.rnCSS.replace(/=/gm, ':') + ';';
    return computedString;
}
const styleMap = {};
function getStyle(hash, style) {
    const styleInfo = styleMap[hash];
    if (styleInfo) {
        styleInfo.usage++;
        return styleInfo.style;
    }
    else {
        const sheet = react_native_1.StyleSheet.create({ [hash]: style });
        return (styleMap[hash] = { style: sheet[hash], usage: 1 }).style;
    }
}
function removeStyle(hash) {
    styleMap[hash].usage--;
    if (styleMap[hash].usage <= 0)
        delete styleMap[hash];
}
const styled = (Component) => {
    const styledComponent = (chunks, ...functs) => {
        const ForwardRefComponent = react_1.default.forwardRef((props, ref) => {
            // const rem = React.useContext(RemContext)
            const shared = react_1.default.useContext(exports.SharedValue);
            // Build the css string with the context
            const css = react_1.default.useMemo(() => buildCSSString(chunks, functs, props, shared), [props, shared]);
            // Store the style in RN format
            const rnStyle = react_1.default.useMemo(() => (0, cssToRN_1.default)(css), [css]);
            const units = react_1.default.useMemo(() => (exports.defaultUnits), [exports.defaultUnits]);
            const { style: styleConvertedFromCSS, hash } = react_1.default.useMemo(() => {
                const style = (0, convertStyle_1.default)(rnStyle, units);
                delete style.textOverflow;
                const hash = (0, generateHash_1.default)(JSON.stringify(style));
                return { style: getStyle(hash, style), hash };
            }, [rnStyle, units]);
            const newProps = react_1.default.useMemo(() => {
                const newProps = { style: [styleConvertedFromCSS, props.style] };
                if (rnStyle.textOverflow === 'ellipsis') {
                    Object.assign(newProps, { numberOfLines: 1 });
                }
                return newProps;
            }, [props.style, rnStyle.textOverflow, styleConvertedFromCSS]);
            react_1.default.useEffect(() => () => removeStyle(hash), [hash]);
            // em !== parentEm alone is a bit dangerous as the component would rerender when the font size change
            return react_1.default.createElement(Component, { ref: ref, ...props, ...newProps });
        });
        return ForwardRefComponent;
    };
    // provide styled(Comp).attrs({} | () => {}) feature
    styledComponent.attrs = (opts) => (chunks, ...functs) => {
        const ComponentWithAttrs = styledComponent(chunks, ...functs);
        const ForwardRefComponent = react_1.default.forwardRef((props, ref) => {
            const attrs = (opts instanceof Function) ? opts(props) : opts;
            return react_1.default.createElement(ComponentWithAttrs, { ref: ref, ...props, ...attrs });
        });
        // TODO : Find a way to remove from the Props the properties affected by opts
        return ForwardRefComponent;
    };
    return styledComponent;
};
exports.default = styled;
const styledFlatList = (chunks, ...functs) => (props) => invoke(styled(react_native_1.FlatList)(chunks, ...functs), props);
exports.styledFlatList = styledFlatList;
exports.styledFlatList.attrs = (opts) => (chunks, ...functs) => (componentProps) => invoke(styled(react_native_1.FlatList).attrs(opts)(chunks, ...functs), componentProps);
const styledSectionList = (chunks, ...functs) => (props) => invoke(styled(react_native_1.SectionList)(chunks, ...functs), props);
exports.styledSectionList = styledSectionList;
exports.styledSectionList.attrs = (opts) => (chunks, ...functs) => (componentProps) => invoke(styled(react_native_1.SectionList).attrs(opts)(chunks, ...functs), componentProps);
const styledVirtualizedList = (chunks, ...functs) => (props) => invoke(styled(react_native_1.VirtualizedList)(chunks, ...functs), props);
exports.styledVirtualizedList = styledVirtualizedList;
exports.styledVirtualizedList.attrs = (opts) => (chunks, ...functs) => (componentProps) => invoke(styled(react_native_1.VirtualizedList).attrs(opts)(chunks, ...functs), componentProps);
function invoke(Component, props) {
    return react_1.default.createElement(Component, { ...props });
}
```

**Comment by Sharcoux** [comment]
If the second version worked, try this one:

```js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.styledVirtualizedList = exports.styledSectionList = exports.styledFlatList = exports.SharedValue = exports.FontSizeContext = exports.RemContext = exports.defaultUnits = void 0;
/* eslint-disable react/prop-types */
/* eslint-disable react/display-name */
const react_1 = __importDefault(require("react"));
const react_native_1 = require("react-native");
const convertStyle_1 = __importDefault(require("./convertStyle"));
const cssToRN_1 = __importDefault(require("./cssToRN"));
const features_1 = require("./features");
const generateHash_1 = __importDefault(require("./generateHash"));
exports.defaultUnits = { em: 16, vw: 1, vh: 1, vmin: 1, vmax: 1, rem: 16, px: 1, pt: 72 / 96, in: 96, pc: 9, cm: 96 / 2.54, mm: 96 / 25.4 };
exports.RemContext = react_1.default.createContext(exports.defaultUnits.rem);
exports.FontSizeContext = react_1.default.createContext(exports.defaultUnits.em);
// We use this to share value within the component (Theme, Translation, whatever)
exports.SharedValue = react_1.default.createContext(undefined);
function buildCSSString(chunks, functs, props, shared) {
    let computedString = chunks.map((chunk, i) => ([chunk, (functs[i] instanceof Function) ? functs[i]({ shared, theme: shared, ...props }) : functs[i]])).flat().join('');
    if (props.rnCSS)
        computedString += props.rnCSS.replace(/=/gm, ':') + ';';
    return computedString;
}
const styleMap = {};
function getStyle(hash, style) {
    const styleInfo = styleMap[hash];
    if (styleInfo) {
        styleInfo.usage++;
        return styleInfo.style;
    }
    else {
        const sheet = react_native_1.StyleSheet.create({ [hash]: style });
        return (styleMap[hash] = { style: sheet[hash], usage: 1 }).style;
    }
}
function removeStyle(hash) {
    styleMap[hash].usage--;
    if (styleMap[hash].usage <= 0)
        delete styleMap[hash];
}
const styled = (Component) => {
    const styledComponent = (chunks, ...functs) => {
        const ForwardRefComponent = react_1.default.forwardRef((props, ref) => {
            const rem = react_1.default.useContext(exports.RemContext);
            const shared = react_1.default.useContext(exports.SharedValue);
            // Build the css string with the context
            const css = react_1.default.useMemo(() => buildCSSString(chunks, functs, props, shared), [props, shared]);
            // Store the style in RN format
            const rnStyle = react_1.default.useMemo(() => (0, cssToRN_1.default)(css), [css]);
            const { needsLayout, needsHover } = react_1.default.useMemo(() => ({
                // needsFontSize: !!css.match(/\b(\d+)(\.\d+)?em\b/)
                // needsScreenSize: !!css.match(/\b(\d+)(\.\d*)?v([hw]|min|max)\b/) || !!rnStyle.media,
                needsLayout: !!css.match(/\d%/),
                needsHover: !!rnStyle.hover
            }), [css, rnStyle.hover]);
            // Handle hover
            const { onMouseEnter, onMouseLeave, hover } = (0, features_1.useHover)(props.onMouseEnter, props.onMouseLeave, needsHover);
            const tempStyle = react_1.default.useMemo(() => {
                const style = { ...rnStyle };
                delete style.media;
                delete style.hover;
                if (hover)
                    Object.assign(style, rnStyle.hover);
                return style;
            }, [hover, rnStyle]);
            // Calculate current em unit for media-queries
            // const parentEm = React.useContext(FontSizeContext)
            // const { em: tempEm } = useFontSize(tempStyle.fontSize, rem, parentEm)
            // Handle layout data needed for % units
            const { width, height, onLayout } = (0, features_1.useLayout)(props.onLayout, needsLayout);
            // Handle screen size needed for vw and wh units
            // const screenUnits = useScreenSize()
            /** The units before re-updating the em */
            const baseUnits = react_1.default.useMemo(() => ({
                ...exports.defaultUnits,
                rem,
                // em: tempEm,
                width,
                height
                // ...screenUnits
            }), [height, rem, width]);
            // apply media queries
            // const mediaQuery = useMediaQuery(rnStyle.media, baseUnits)
            // Handle em units
            // const fontSize = (mediaQuery && mediaQuery.fontSize) || tempStyle.fontSize
            // const { em } = useFontSize(fontSize, baseUnits.rem, parentEm)
            const finalStyle = react_1.default.useMemo(() => {
                const style = { ...tempStyle };
                if (style.fontSize)
                    style.fontSize = 16 + 'px';
                if (style.zIndex === undefined && react_native_1.Platform.OS === 'web')
                    style.zIndex = 'auto';
                return style;
            }, [tempStyle]);
            const units = react_1.default.useMemo(() => ({ ...baseUnits }), [baseUnits]);
            const { style: styleConvertedFromCSS, hash } = react_1.default.useMemo(() => {
                const style = (0, convertStyle_1.default)(finalStyle, units);
                delete style.textOverflow;
                const hash = (0, generateHash_1.default)(JSON.stringify(style));
                return { style: getStyle(hash, style), hash };
            }, [finalStyle, units]);
            const newProps = react_1.default.useMemo(() => {
                const newProps = { style: [styleConvertedFromCSS, props.style], onMouseEnter, onMouseLeave, onLayout };
                if (finalStyle.textOverflow === 'ellipsis') {
                    Object.assign(newProps, { numberOfLines: 1 });
                }
                return newProps;
            }, [finalStyle.textOverflow, onLayout, onMouseEnter, onMouseLeave, props.style, styleConvertedFromCSS]);
            react_1.default.useEffect(() => () => removeStyle(hash), [hash]);
            // em !== parentEm alone is a bit dangerous as the component would rerender when the font size change
            return react_1.default.createElement(Component, { ref: ref, ...props, ...newProps });
        });
        return ForwardRefComponent;
    };
    // provide styled(Comp).attrs({} | () => {}) feature
    styledComponent.attrs = (opts) => (chunks, ...functs) => {
        const ComponentWithAttrs = styledComponent(chunks, ...functs);
        const ForwardRefComponent = react_1.default.forwardRef((props, ref) => {
            const attrs = (opts instanceof Function) ? opts(props) : opts;
            return react_1.default.createElement(ComponentWithAttrs, { ref: ref, ...props, ...attrs });
        });
        // TODO : Find a way to remove from the Props the properties affected by opts
        return ForwardRefComponent;
    };
    return styledComponent;
};
exports.default = styled;
const styledFlatList = (chunks, ...functs) => (props) => invoke(styled(react_native_1.FlatList)(chunks, ...functs), props);
exports.styledFlatList = styledFlatList;
exports.styledFlatList.attrs = (opts) => (chunks, ...functs) => (componentProps) => invoke(styled(react_native_1.FlatList).attrs(opts)(chunks, ...functs), componentProps);
const styledSectionList = (chunks, ...functs) => (props) => invoke(styled(react_native_1.SectionList)(chunks, ...functs), props);
exports.styledSectionList = styledSectionList;
exports.styledSectionList.attrs = (opts) => (chunks, ...functs) => (componentProps) => invoke(styled(react_native_1.SectionList).attrs(opts)(chunks, ...functs), componentProps);
const styledVirtualizedList = (chunks, ...functs) => (props) => invoke(styled(react_native_1.VirtualizedList)(chunks, ...functs), props);
exports.styledVirtualizedList = styledVirtualizedList;
exports.styledVirtualizedList.attrs = (opts) => (chunks, ...functs) => (componentProps) => invoke(styled(react_native_1.VirtualizedList).attrs(opts)(chunks, ...functs), componentProps);
function invoke(Component, props) {
    return react_1.default.createElement(Component, { ...props });
}
```

**Comment by TeixeiraFillipe** [comment]
Tried to replace the content of the file as you asked, however it didn't work with any of them. It shows the black screen in release just by importing the lib, we don't even have to use, just its declaration is enough to break.

**Comment by Sharcoux** [comment]
Ok, interesting. Well, the first file was basically removing all of the code of rn-css, so this checks a lot of things.

I need you to do try this now:
- Replace the styleComponent.js file with the **first** one I provided
- Replace the file `rn-css/dist/index.js` with this:

```js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RemContext = exports.FontSizeContext = exports.SharedValue = void 0;
const RN = __importStar(require("react-native"));
const styleComponent_1 = __importStar(require("./styleComponent"));
// export { cssToRNStyle } from './cssToRN'
var styleComponent_2 = require("./styleComponent");
Object.defineProperty(exports, "SharedValue", { enumerable: true, get: function () { return styleComponent_2.SharedValue; } });
Object.defineProperty(exports, "FontSizeContext", { enumerable: true, get: function () { return styleComponent_2.FontSizeContext; } });
Object.defineProperty(exports, "RemContext", { enumerable: true, get: function () { return styleComponent_2.RemContext; } });
// export * from './useTheme'
const styled = (Component) => (0, styleComponent_1.default)(Component);
styled.ActivityIndicator = styled(RN.ActivityIndicator);
styled.Button = styled(RN.Button);
styled.DrawerLayoutAndroid = styled(RN.DrawerLayoutAndroid);
styled.Image = styled(RN.Image);
styled.ImageBackground = styled(RN.ImageBackground);
styled.KeyboardAvoidingView = styled(RN.KeyboardAvoidingView);
styled.Modal = styled(RN.Modal);
styled.NavigatorIOS = styled(RN.NavigatorIOS);
styled.ScrollView = styled(RN.ScrollView);
styled.SnapshotViewIOS = styled(RN.SnapshotViewIOS);
styled.Switch = styled(RN.Switch);
styled.RecyclerViewBackedScrollView = styled(RN.RecyclerViewBackedScrollView);
styled.RefreshControl = styled(RN.RefreshControl);
styled.SafeAreaView = styled(RN.SafeAreaView);
styled.StatusBar = styled(RN.StatusBar);
styled.Text = styled(RN.Text);
styled.TextInput = styled(RN.TextInput);
styled.TouchableHighlight = styled(RN.TouchableHighlight);
styled.TouchableNativeFeedback = styled(RN.TouchableNativeFeedback);
styled.TouchableOpacity = styled(RN.TouchableOpacity);
styled.TouchableWithoutFeedback = styled(RN.TouchableWithoutFeedback);
styled.View = styled(RN.View);
styled.FlatList = styleComponent_1.styledFlatList;
styled.SectionList = styleComponent_1.styledSectionList;
styled.VirtualizedList = styleComponent_1.styledVirtualizedList;
exports.default = styled;
```

**Comment by TeixeiraFillipe** [comment]
It didn't work either, unfortunately. I'm trying to remove some parts of the lib's code as well so I can isolate what's not working. 
If I find anything, I'll let you know. If you have any ideas of anything I can try, I would be glad to try.

**Comment by Sharcoux** [comment]
Well, just to be sure, try to replace the index.js with just this:

```js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const styled = (Component) => Component;
exports.default = styled;
```

**Comment by TeixeiraFillipe** [comment]
Like I said above, just by importing rn-css, I get the black screen. However, when I replace with the code above, when just importing (not creating any component), it works, but if I try to create a component with rn-css, it shows the black screen again.

**Comment by TeixeiraFillipe** [comment]
I was changing some code lines from the files of the lib to try to isole the issue, and I found out that the issue happens with regular expressions. Not sure why, however the release build shows a black screen when using regular expressions. The file I had issues with were `rn-css/dist/cssToRN/mediaQueries.js` and `rn-css/dist/cssToRN/index.js`. Still thinking of a way to fix this, though, since it seems to be an issue with react native Windows.

If you have any ideas, let me know, please.

**Comment by Sharcoux** [comment]
Might be matchAll or lookahead. You can try to add a polyfill and remove the lookahead in cssChunkToStyle.

**Comment by Sharcoux** [comment]
I'll be interested to know which of those is causing the issue. Maybe you could try to build without rn-css and try to add a regex with a lookahead, and then a build using matchAll to see which one is causing the issue?

**Comment by TeixeiraFillipe** [comment]
Hey, sorry for the delay. Just tried to build in release using matchAll and it didn't work. Lookahead seems to be fine.

**Comment by Sharcoux** [comment]
Got it. Then I know what to do to solve this. Thanks for your help. I'll come back with a fix.

**Comment by Sharcoux** [comment]
Please try to install version `1.6.14-0` of rn-css and tell me if it fixes your issue. I added polyfill for matchAll and flat methods.

`npm i -D rn-css@1.6.14-0`

**Comment by TeixeiraFillipe** [comment]
Hey, thank you for the polyfill! It works for the matchAll and the flat functions, however, just installing this version doesn't work on Windows due to the 'dotall' flags used in some regex, such as the 'mediaQueries.js' file. Removing these flags, it works!

**Comment by Sharcoux** [comment]
Try this one:

`npm i -D rn-css@1.6.14-1`

**Comment by Sharcoux** [comment]
`npm i -D rn-css@1.6.14-2`?

**Comment by TeixeiraFillipe** [comment]
Thank you François! This fixed all the issues and we were able to build just fine in Windows release. Thank you!

**Comment by Sharcoux** [comment]
Fixed in 1.6.14
Thanks a lot for your help.
